pkg/upapi/error.go

```diff
@@ -0,0 +1,62 @@
+package upapi
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"strconv"
+)
+
+var DecodeError = errors.New("error response decode error")
+
+type Error struct {
+	Response *http.Response
+	Code     string              `json:"error_code"`
+	Message  string              `json:"error_message"`
+	Fields   map[string][]string `json:"error_fields,omitempty"`
+}
+
+func NewError() *Error {
+	return &Error{
+		Fields: make(map[string][]string),
+	}
+}
+
+func (e Error) Error() string {
+	if e.Fields == nil {
```

Comment:
maybe you can use a `strings.Builder` here?

Reply:
Good point.